loki/transform/dependency_transform.py

```diff
@@ -94,10 +96,29 @@ def transform_subroutine(self, routine, **kwargs):
         imports = FindNodes(Import).visit(routine.ir)
         self.rename_imports(routine, imports=imports, **kwargs)
 
+        # Interface blocks can only be in the spec
+        intfs = FindNodes(Interface).visit(routine.spec)
+        self.rename_interfaces(routine, intfs=intfs, **kwargs)
+
         if role == 'kernel' and self.mode == 'strict':
             # Re-generate C-style interface header
             self.generate_interfaces(routine)
 
+    def update_result_var(self, routine):
+        """
+        Update name of result variable for function calls.
+        """
```

Comment:
Spotted this only now. Does this work as expected with functions that are declared with a different result name?

```f90
function my_func result(ret)
    integer ret
    ret = 5
end function my_func
```

Reply:
Ah, great catch! You are right, it won't work. It needs a test to check if `result_name` matches `Subroutine.name`. I'll update and add a test 👍

Reply:
Sorry, did not understand the fix, but cleared up offline. All good now